### ISSUE TITLE
Forms list data sorting is broken

### ISSUE THREAD

**Issue by civing** [Lundalogik/lime-elements#1621]
## Current behavior

https://user-images.githubusercontent.com/4279390/154450313-4ea1bd9b-5d6d-48c3-8427-23f400d741c5.mov

The sorting arrows does not seem to switch place following a sort.

1. Go to the "List data" example for limel forms https://lundalogik.github.io/lime-elements/versions/NEXT-35.0.0-next.24/#/component/limel-form/
2. Click the sorting down arrow associated with the entry "Joker"
3. Now click the Click the sorting down arrow associated with the entry "Joker" again

## Expected behavior
Clicking the sorting arrow associated with the entry should always move the entry on the same row.

## Environment

- **lime-elements** version: NEXT-35.0.0-next.24
- Framework used: Don't know
- Logs:

**Comment by adrianschmidt** [comment]
Those are not for sorting the list the way you sort a table, but for moving the item they are attached to up or down in the list. I agree that the feedback to the user about what is happening is far from ideal though… 😞

**Comment by adrianschmidt** [comment]
@Kiarokh I guess, ideally, we'd like to have a little animation of the card moving to its new place or something. But I imagine that would take a bit of work… is there some easier way we could do _something_ to improve the feedback telling the user what's happening here?

**Comment by Kiarokh** [comment]
I agree with animations @adrianschmidt it would certainly help. But I think what @civing is showing here is really broken 😞 

The video shows:
Click ⬇️ on Joker, it goes down one row. Click ⬇️  again on Joker (row 2), and joker is supposed to go to row 3, but it goes back on row 2 again. And the next flow he shows is also problematic. The arrows are not really representing the row they belong to.

**Comment by Kiarokh** [comment]
and add one more item, and it'll not work at all

**Comment by Kiarokh** [comment]
Do you know an example of where this is used? 
I know we are using the one with collapsible section in Admin, 👇 
![image](https://user-images.githubusercontent.com/35954987/154516921-7bbb3feb-7802-4450-bf7b-577b65230fe0.png)

but not sure where we use the other ones. Maybe @jgroth knows

**Comment by jgroth** [comment]
> I know we are using the one with collapsible section in Admin, point_down and they're working fine

Oh, if it's working fine there maybe the example is just broken? :thinking:

**Comment by adrianschmidt** [comment]
In Lime Admin we're using collapsible-sections, but here it's input fields, and the icons are in a different order in the input field case too, so there's some difference between them. I think we added Stencil's `key` attribute to the collapsible-sections a while back, to make the `open` state follow the content. My first attempt at a solution would be doing the same here.
